bagitobjecttransfer/recordtransfer/tests/unit/test_views.py

```diff
@@ -701,3 +708,180 @@ def test_empty_form_submission(self):
+        response = self.client.post(self.url, data=form_data, HTTP_REFERER="transfer")
+        self.assertEqual(response.status_code, 200)
+        response_json = response.json()
+        self.assertEqual(response_json["message"], gettext("Group created"))
+        self.assertEqual(response_json["status"], "success")
+
+    @patch("recordtransfer.views.SubmissionGroupCreateView.form_invalid")
+    def test_form_invalid_json_response(self, mock_form_invalid: MagicMock) -> None:
+        """Test that a JsonResponse is returned when the form is invalid and submitted from the
+        transfer page.
+        """
+        mock_form_invalid.return_value = JsonResponse(
+            {"message": "This field is required.", "status": "error"}, status=400
+        )
+        form_data = {
+            "name": "",
+            "description": "Test Description",
+        }
+        response = self.client.post(self.url, data=form_data, HTTP_REFERER="transfer")
+        self.assertEqual(response.status_code, 400)
+        response_json = response.json()
+        self.assertEqual(response_json["message"], "This field is required.")
+        self.assertEqual(response_json["status"], "error")
+
+
+class TestSubmissionGroupDetailView(TestCase):
+    """Tests for SubmissionGroupDetailView."""
+
+    @classmethod
+    def setUpTestData(cls) -> None:
+        """Set up test data."""
+        cls.user = User.objects.create_user(username="testuser", password="password")
+        cls.staff_user = User.objects.create_user(
+            username="staffuser", password="password", is_staff=True
+        )
+        cls.group = SubmissionGroup.objects.create(
+            name="Test Group", description="Test Description", created_by=cls.user
+        )
+        cls.url = reverse("recordtransfer:submissiongroupdetail", kwargs={"uuid": cls.group.uuid})
+
+    def setUp(self) -> None:
+        """Set up test environment."""
+        self.client.login(username="testuser", password="password")
+
+    def test_access_authenticated_user(self) -> None:
+        """Test that an authenticated user can access the view."""
+        response = self.client.get(self.url)
+        self.assertEqual(response.status_code, 200)
+        self.assertTemplateUsed(response, "recordtransfer/submission_group_show_create.html")
+
+    def test_access_unauthenticated_user(self) -> None:
+        """Test that an unauthenticated user is redirected to the login page."""
+        self.client.logout()
+        response = self.client.get(self.url)
+        self.assertRedirects(response, f"{reverse('login')}?next={self.url}")
+
+    def test_access_non_creator_user(self) -> None:
+        """Test that a non-creator user cannot access the view."""
+        non_creator_user = User.objects.create_user(username="noncreator", password="password")
+        self.client.login(username="noncreator", password="password")
+        response = self.client.get(self.url)
+        self.assertEqual(response.status_code, 403)
```

Comment:
This is an interesting case you've found here. I think the view should actually return 404 in this test. 403 implies the resource exists, but the user can't access it. I would prefer the user to be given a 404 so that they don't even know whether that's a valid resource URL (for security purposes).